index.js

```diff
@@ -413,16 +413,19 @@ module.exports = function(app) {
 
   function saveState() {
     let pdState = {}
-    for (let i = 1; i <= virtualSwitch.length; i++) {
+    for (let i = 1; i <= Object.keys(virtualSwitch).length; i++) {
       pdState[i] = virtualSwitch[i].state === 1 ? "ON" : "OFF"
     }
 
     let filepath = app.getDataDirPath() + pdStateSuffix
 
     app.debug('Writing ' + JSON.stringify(pdState) + ' to file ' + filepath)
-    fs.writeFile(filepath, JSON.stringify(pdState, null, 2), (err) => {
-      if (err) app.error('Could not write to ' + filepath + ' ERROR: ' + err)
-    })
+
+    try {
```

Comment:
Need to write this synchronously. On plugin restart, the read can occur before an asynchronous write is finished.